homeassistant/core.py

```diff
@@ -1175,12 +1175,9 @@ def _async_listen_filterable_job(
         self, event_type: str, filterable_job: _FilterableJobType
     ) -> CALLBACK_TYPE:
         self._listeners.setdefault(event_type, []).append(filterable_job)
-
-        def remove_listener() -> None:
-            """Remove the listener."""
-            self._async_remove_listener(event_type, filterable_job)
-
-        return remove_listener
+        return functools.partial(
```

Comment:
Are there good sources online for the performance difference?

Reply:
Probably not.  There is a benchmark script I put in a prior PR for a similar change though.  I can find it when I get back to the house in a few hours

Reply:
Looks like that was a memory test, which is still good for this either way.

When we switched to using partial for #93602 I remember there startup time improvement as well.

```
Memory used per closure: 408.7808
Memory used per partial: 250.01984
```

```python
from functools import partial

import psutil

process = psutil.Process()
before = process.memory_info().rss
closures = []


def gen_closure(i, y, q):
    def this_is_a_closure():
        z = i + 1
        print(z)
        z = y + 1
        print(z)
        z += q
        new_dict = {}
        new_dict["key"] = "value"
        return i

    return this_is_a_closure


for i in range(100000):
    closures.append(gen_closure(i, 6, 7))

after = process.memory_info().rss

print(f"Memory used per closure: {(after - before) / len(closures):,}")


before = process.memory_info().rss


def use_for_partial(i, y, q):
    z = i + 1
    print(z)
    z = y + 1
    print(z)
    z += q
    new_dict = {}
    new_dict["key"] = "value"
    return i


partials = []
for i in range(100000):
    partials.append(partial(use_for_partial, i, 6, 7))

after = process.memory_info().rss
print(f"Memory used per partial: {(after - before) / len(partials):,}")
```

Reply:
partial creates a new function wrapper with different arguments but the underlying function body reference is the same which avoids the overhead of constructing a whole new function every time.

https://github.com/python/cpython/blob/cf19e8ea3a232086ff1e0d7d5e2a092d3d96fc7c/Modules/_functoolsmodule.c#L79
https://github.com/python/cpython/blob/cf19e8ea3a232086ff1e0d7d5e2a092d3d96fc7c/Modules/_functoolsmodule.c#L101

Reply:
Ok.